projects/ziti-console-lib/src/lib/pages/services/services-page.service.ts

```diff
@@ -0,0 +1,198 @@
+import {DataTableFilterService, FilterObj} from "../../features/data-table/data-table-filter.service";
+import {ListPageServiceClass} from "../../shared/list-page-service.class";
+import {
+    TableColumnDefaultComponent
+} from "../../features/data-table/column-headers/table-column-default/table-column-default.component";
+import {CallbackResults} from "../../features/list-page-features/list-page-form/list-page-form.component";
+import {SETTINGS_SERVICE, SettingsService} from "../../services/settings.service";
+import {ZITI_DATA_SERVICE, ZitiDataService} from "../../services/ziti-data.service";
+import {CsvDownloadService} from "../../services/csv-download.service";
+import {Service} from "../../models/service";
+import {unset} from "lodash";
+import {GrowlerModel} from "../../features/messaging/growler.model";
+import {GrowlerService} from "../../features/messaging/growler.service";
+import {MatDialog} from "@angular/material/dialog";
+import {SettingsServiceClass} from "../../services/settings-service.class";
+
+const CSV_COLUMNS = [
+    {label: 'Name', path: 'name'},
+    {label: 'Created At', path: 'createdAt'}
+];
+
+@Injectable({
+    providedIn: 'root'
+})
+export class ServicesPageService extends ListPageServiceClass {
+
+    private paging = this.DEFAULT_PAGING;
+    public sideModalOpen = false;
+    public modalType = 'service';
+
+    selectedService: any = new Service();
+    columnFilters: any = {
+        name: '',
+        os: '',
+        createdAt: '',
+    };
+
+    override menuItems = [
+        {name: 'Edit', action: 'update'},
+        {name: 'Delete', action: 'delete'},
+    ]
+
+    override tableHeaderActions = [
+        {name: 'Download All', action: 'download-all'},
+        {name: 'Download Selected', action: 'download-selected'},
+    ]
+
+    constructor(
+        @Inject(SETTINGS_SERVICE) settings: SettingsServiceClass,
+        filterService: DataTableFilterService,
+        @Inject(ZITI_DATA_SERVICE) private zitiService: ZitiDataService,
+        override csvDownloadService: CsvDownloadService,
+        private growlerService: GrowlerService,
+        private dialogForm: MatDialog,
+    ) {
+        super(settings, filterService, csvDownloadService);
+    }
+
+    validate = (formData): Promise<CallbackResults> => {
+        return Promise.resolve({ passed: true});
+    }
+
+    initTableColumns(): any {
+        const nameRenderer = (row) => {
+            return `<div class="col cell-name-renderer" data-id="${row?.data?.id}">
+                <strong>${row?.data?.name}</strong>
+              </div>`
+        }
+
+        const rolesRenderer = (row) => {
+            let roles = '';
+            row?.data?.roleAttributes?.forEach((attr) => {
+                roles += '<div class="hashtag">'+attr+'</div>';
+            });
+            return roles;
+        }
+
+        const createdAtFormatter = (row) => {
+            return moment(row?.data?.createdAt).local().format('M/D/YYYY H:MM A');
```

Comment:
can we set this up for internationalization? this is most definitely american-centric...